src/braket/circuits/text_diagram_builders/ascii_circuit_diagram.py

```diff
@@ -0,0 +1,176 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+from functools import reduce
+from typing import Union
+
+import braket.circuits.circuit as cir
+from braket.circuits.circuit_diagram import CircuitDiagram
+from braket.circuits.compiler_directive import CompilerDirective
+from braket.circuits.gate import Gate
+from braket.circuits.instruction import Instruction
+from braket.circuits.result_type import ResultType
+from braket.circuits.text_diagram_builders import text_circuit_diagram_utils
+from braket.registers.qubit_set import QubitSet
+
+
+class AsciiCircuitDiagram(CircuitDiagram):
+    """Builds ASCII string circuit diagrams."""
+
+    _vdelim = "|"  # Character that connects qubits of multi-qubit gates
```

Comment:
These classes seem to have a mix of functionality and configuration and create a (soft) cyclic dependency. I think we can do either one or the other, and avoid the cycle. 

So, either:
 - we have a configuration that we pass into diagram_utils, and have it do the rest of the work based on the configuration
XOR
 - we have the circuit class call the diagram utils and pass in the diagram classes and have them handle the functionality where the configuration is relevant. 
 
I think it's confusing to have both.